spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -655,17 +655,11 @@ else if (getApplicationContext() != null) {
 		}
 		messageListenerContainer.setBeanName(destination + ".container");
 		// end of these won't be needed...
-		if (!extendedConsumerProperties.getExtension().isAutoCommitOffset()) {
-			messageListenerContainer.getContainerProperties()
-					.setAckMode(ContainerProperties.AckMode.MANUAL);
-			messageListenerContainer.getContainerProperties().setAckOnError(false);
-		}
-		else {
-			messageListenerContainer.getContainerProperties()
-					.setAckOnError(isAutoCommitOnError(extendedConsumerProperties));
-			if (extendedConsumerProperties.getExtension().isAckEachRecord()) {
+		final ContainerProperties.AckMode ackMode = extendedConsumerProperties.getExtension().getAckMode();
+		if (!extendedConsumerProperties.isBatchMode()) { //if batch mode, we will ignore any ack mode.
```

Comment:
We need to allow ack modes other than RECORD with batch mode; it's just that RECORD mode won't work with batchMode so that should be ignored.